django/forms/forms.py

```diff
@@ -252,7 +252,7 @@ def non_field_errors(self):
             self.error_class(error_class="nonfield", renderer=self.renderer),
         )
 
-    def add_error(self, field, error):
+    def add_error(self, field, error, field_id=None):
```

Comment:
I feel like we could do something like:
```diff
--- a/django/forms/forms.py
+++ b/django/forms/forms.py
@@ -252,7 +252,7 @@ class BaseForm(RenderableFormMixin):
             self.error_class(error_class="nonfield", renderer=self.renderer),
         )
 
-    def add_error(self, field, error, field_id=None):
+    def add_error(self, field, error):
         """
         Update the content of `self._errors`.
 
@@ -298,6 +298,7 @@ class BaseForm(RenderableFormMixin):
                         error_class="nonfield", renderer=self.renderer
                     )
                 else:
+                    field_id = self.__getitem__(field).auto_id
                     self._errors[field] = self.error_class(
                         renderer=self.renderer,
                         field_id=field_id,
@@ -338,7 +339,7 @@ class BaseForm(RenderableFormMixin):
                     value = getattr(self, "clean_%s" % name)()
                     self.cleaned_data[name] = value
             except ValidationError as e:
-                self.add_error(name, e, bf.auto_id)
+                self.add_error(name, e)
 
     def _clean_form(self):
```

If we don't do this, we need doc updates to `.. method:: Form.add_error` and anyone who has used `.add_error` will be missing the id I suppose